src/datastore_mad/remotes/lxd_downloader.sh

```diff
@@ -0,0 +1,161 @@
+    k=${kvp[0]};v=${kvp[1]};
+    eval $k=$v;
+done
+
+rootfs_url=`echo $url | grep -oP "^"lxd://"\K.*"`
+curl="curl -L"
+if [[ $rootfs_url == *"turnkeylinux.org"* ]]; then
+    untar_options="xvzpf"
+    extension="tar.gz"
+    distro=`echo $rootfs_url | cut -d '-' -f 1 | cut -d '/' -f 7`
+    version_number=`echo $rootfs_url | cut -d '-' -f 2`
+    if [[ $version_number == "9" ]]; then
+        version="Stretch"
+    elif [[ $version_number == "8" ]]; then
+        version="Jessie"
+    else
+        version="undefined"
+    fi
+    release_date=""
+elif [[ $rootfs_url == *"linuxcontainers.org"* ]]; then
+    untar_options="xvJpf"
+    extension="tar.xz"
+    distro=`echo $rootfs_url | cut -d '/' -f 5`
+    version=`echo $rootfs_url | cut -d '/' -f 6`
+    release_date=`echo $rootfs_url | cut -d '/' -f 9`
+fi
+
+output=$tmp_dir/$id.$extension
+
+$curl $rootfs_url --output $output --silent
+qemu-img create -f raw $tmp_dir/$id.raw ${size}M  > /dev/null 2>&1
+
+case $filesystem in
+    "ext4")
+        mkfs.ext4 -F $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+    "xfs")
+        mkfs.xfs -f $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+    *)
+        mkfs.ext4 -F $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+esac
+
+
+mkdir $tmp_dir/$id
+sudo mount $tmp_dir/$id.raw $tmp_dir/$id
+sudo chown oneadmin:oneadmin $tmp_dir/$id
+sudo tar $untar_options $output -C $tmp_dir/$id > /dev/null 2>&1
+sync
+
+context_url="https://github.com/OpenNebula/addon-context-linux/releases/download"
+
+case "$rootfs_url" in
+*ubuntu*|*debian*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        export PATH=$PATH:/bin:/sbin
+        rm -f /etc/resolv.conf > /dev/null 2>&1
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        apt-get update > /dev/null
+        apt-get install curl -y > /dev/null 2>&1
+        $curl $context_url/v$selected_tag/one-context_$selected_tag-1.deb -Lsfo /root/context.deb
+        apt-get install /root/context.deb -y > /dev/null 2>&1
+EOT
+    ;;
+*centos/6*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        export PATH=$PATH:/bin:/sbin
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el6.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
+EOT
+    ;;
+*centos/7*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el7.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
+EOT
+    ;;
+*alpine*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el7.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
```

Comment:
Alpine is `apk add --allow-untrusted ...`

Reply:
Thanks! Fixed.